File: samples/pipeline/python/infra/pipeline.py
import os
from aws_cdk import App, Environment
from pipeline_stack import PipelineStack
from application_stage import ApplicationStage
from aws_prototyping_sdk.pdk_nag import PDKNag
from aws_prototyping_sdk.pipeline import PDKPipeline



app = PDKNag.app()

branch_prefix = PDKPipeline.get_branch_prefix(node=app.node)

pipelineStack = PipelineStack(app, branch_prefix + "PipelineStack", env=Environment(
    account=os.environ["CDK_DEFAULT_ACCOUNT"],
    region=os.environ["CDK_DEFAULT_REGION"]
))

devStage = ApplicationStage(app, branch_prefix + "Dev", env=Environment(
    account=os.environ["CDK_DEFAULT_ACCOUNT"],
    region=os.environ["CDK_DEFAULT_REGION"]
))

pipelineStack.pipeline.add_stage(devStage)

# Add additional stages here i.e. Prod
# if PDKPipeline.is_default_branch(node=app.node): ...

pipelineStack.pipeline.build_pipeline()  # Needed for CDK Nag
app.synth()

File: samples/pipeline/java/src/main/java/software/aws/Pipeline.java
package software.aws;

import software.aws.awsprototypingsdk.pdknag.PDKNagApp;
import software.aws.awsprototypingsdk.pdknag.PDKNag;
import software.aws.awsprototypingsdk.pipeline.PDKPipeline;
import software.aws.awsprototypingsdk.pipeline.IsDefaultBranchProps;
import software.amazon.awscdk.Environment;
import software.amazon.awscdk.StackProps;
import software.amazon.awscdk.StageProps;

public class Pipeline {
    public static void main(String[] args) {
        PDKNagApp app = PDKNag.app();

        String branchPrefix = PDKPipeline.getBranchPrefix(IsDefaultBranchProps.builder()
                .node(app.getNode())
                .build());

        PipelineStack pipelineStack = new PipelineStack(app, branchPrefix + "PipelineStack", StackProps.builder()
                .env(Environment.builder()
                        .account(System.getenv("CDK_DEFAULT_ACCOUNT"))
                        .region(System.getenv("CDK_DEFAULT_REGION"))
                        .build())
                .build());

        ApplicationStage devStage = new ApplicationStage(app, branchPrefix + "Dev", StageProps.builder()
                .env(Environment.builder()
                        .account(System.getenv("CDK_DEFAULT_ACCOUNT"))
                        .region(System.getenv("CDK_DEFAULT_REGION"))
                        .build())
                .build());

        pipelineStack.getPipeline().addStage(devStage);

        // Add additional stages here i.e. Prod
        // if (PDKPipeline.isDefaultBranch(IsDefaultBranchProps.builder()
        //         .node(app.getNode())
        //         .build())) ...

        pipelineStack.getPipeline().buildPipeline(); // Needed for CDK Nag
        app.synth();
    }
}